File: app/src/main/java/com/travlog/android/apps/libs/qualifiers/RequiresActivityViewModel.java
package com.travlog.android.apps.libs.qualifiers;

import com.travlog.android.apps.libs.ActivityViewModel;

import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Inherited
@Retention(RetentionPolicy.RUNTIME)
public @interface RequiresActivityViewModel {
    Class<? extends ActivityViewModel> value();
}

File: app/src/main/java/com/travlog/android/apps/libs/qualifiers/RequiresActivityViewModel.kt
package com.travlog.android.apps.libs.qualifiers

import com.travlog.android.apps.libs.ActivityViewModel
import java.lang.annotation.Inherited
import kotlin.reflect.KClass

@Inherited
@Retention(AnnotationRetention.RUNTIME)
annotation class RequiresActivityViewModel(val value: KClass<out ActivityViewModel<*>>)
